### PR TITLE
normalizer_lib as class

### PR DESCRIPTION
**Pull request by zupon**
This PR turns `normalizer_lib` into a class. This allows us to create a `NormalizerLib` object when we run `normalizer.py` and pass the language as an argument. This will load the appropriate config, along with the language-specific graphemes/numerals/punctuation and language-specific rewrite rules.

@crewee , @sandyritchie showed me something similar that you had worked on that used the `@property` decorator on some class methods. In that case you didn't need to pass any arguments to the methods and in the end can just do something like `string @ self.remove_whitespace @ detach_punctuation ...` rather than the nested functions I have here. I haven't figured that out yet, but I would love to include it here to keep things a bit tidier if you have any tips!

### CODE REVIEW

File: normalizer_lib.py
Comment (crewee) [inline normalizer_lib.py]: I don't think you need return self, I think you can just return self.language_specific_preprocessing.

**Review by crewee — APPROVED** [review]
Looks great!  Let me know if you need the advice on @property.  It looks here like everythings pretty clean!

LGTM, but let's wait for Sandy's approval too.

File: normalizer_lib.py
Comment (zupon) [reply normalizer_lib.py]: Fixed by using `DUMMY_REWRITE`!

File: normalizer_lib.py
Comment (sandyritchie) [inline normalizer_lib.py]: Are these .plus inside the statement redundant?

File: normalizer_lib.py
Comment (sandyritchie) [inline normalizer_lib.py]: Not sure if this need to be a method, you could also just define it in the init of the class and call it where you need it.

File: normalizer_lib.py
Comment (sandyritchie) [inline normalizer_lib.py]: Some repetition from the previous method, consider adding a helper function which contains the shared logic.

File: normalizer_lib.py
Comment (sandyritchie) [inline normalizer_lib.py]: Again doesn't really need to be a method.

File: normalizer_lib.py
Comment (sandyritchie) [inline normalizer_lib.py]: Note that Python's lower() method might not work for all scripts, it's probably okay for our purposes but maybe note it in the docstring.

File: normalizer_lib.py
Comment (sandyritchie) [inline normalizer_lib.py]: Repeated logic could be in a helper method.

File: normalizer_lib.py
Comment (zupon) [reply normalizer_lib.py]: I believe they are.

File: normalizer_lib.py
Comment (zupon) [reply normalizer_lib.py]: I ended up not needing `valid_token`, since that is pretty easily defined by `self.verbalizable()`, but I also put the splitting into tokens in a helper function to tidy things up.

File: normalizer_lib.py
Comment (zupon) [reply normalizer_lib.py]: Good point. I refactored things a bit and put the FSTs in the class init, so now none of those are methods.

File: normalizer_lib.py
Comment (zupon) [reply normalizer_lib.py]: Agreed! See my comment above about moving all the FSTs into the class init.

File: normalizer_lib.py
Comment (zupon) [reply normalizer_lib.py]: I made a helper function `preprocess_string` to get rid of repeated logic.

File: normalizer_lib.py
Comment (zupon) [reply normalizer_lib.py]: Added to the docstring for the new `preprocess_string` helper method where `.lower()` is called.

File: normalizer_lib.py
Comment (sandyritchie) [inline normalizer_lib.py]: Put the strip() call in the helper too.

**Review by sandyritchie — APPROVED** [review]
LGTM!

File: normalizer_lib.py
Comment (zupon) [reply normalizer_lib.py]: Moving it there didn't actually work, since it's more spaces might still exist after the `remove_extra_whitespace` step. However, I changed the right context of that deletion from `byte.SPACE` to `union(byte.SPACE, "[EOS]")`, so it will now strip all extra whitespace at the end of a line.

File: normalizer_lib.py
Comment (crewee) [inline normalizer_lib.py]: Can we add the returns annotation to this line?

def verbalizable(self) -> Fst:

File: normalizer_lib_test.py
Comment (crewee) [inline normalizer_lib_test.py]: This is totally a nit, but this is in CAPS case here, but lower case in "normalizer.py".  Should that be the same?  :)

**Review by crewee — APPROVED** [review]
LGTM after nit comments :)

File: normalizer_lib.py
Comment (zupon) [reply normalizer_lib.py]: Done!

File: normalizer_lib_test.py
Comment (zupon) [reply normalizer_lib_test.py]: The beauty of pylint. Pylint thinks `NORM` here is a constant, so it should be in CAPS. In `normalizer.py` `norm` pytlint thinks it's a variable, so it should be snake_case. I'm all for ignoring pylint's suggestions though in this case if it adds consistency :)

Comment (crewee) [reply normalizer_lib_test.py]: Haha, thanks Pylint!  I think consistency is a good way to go :).  I'll leave it up to you!

Comment (zupon) [reply normalizer_lib_test.py]: I decided to lowercase this one to match `normalizer.py`.

**Pull request merged**
